pilot/pkg/xds/delta.go

```diff
@@ -446,7 +446,7 @@ func (s *DiscoveryServer) pushDeltaXds(con *Connection, push *model.PushContext,
 		Nonce:             nonce(push.LedgerVersion),
 		Resources:         res,
 	}
-	if usedDelta {
+	if len(deletedRes) > 0 {
```

Comment:
We can't rely on the length of `DeletedResources` to determine whether or not delta was used. The case could just be that delta generation was used, but nothing was deleted.

Reply:
But what we care about is sending RemovedResources in Delta Response right? Or do we care about some thing else also?

Reply:
you need a way to distinguish "these are the SotW" and "I generated some new partial updates, with no deletes"
The problem is many generators (including cds depending on the update) return SotW style so we need to convert that to delta (ie determine what were deleted, as done here)

Reply:
Ok. Is this needed for intermediate state? And if for example cds completely moves to delta, we can get rid of this flag?